src/ckeditor.jsx

```diff
@@ -58,6 +58,10 @@ class CKEditor extends React.Component {
 		getEditorNamespace( CKEditor.editorUrl ).then( CKEDITOR => {
 			const constructor = this.props.type === 'inline' ? 'inline' : 'replace';
 
+			if ( this.props.onBeforeLoad ) {
+				this.props.onBeforeLoad( CKEDITOR );
```

Comment:
It would be safer to assume that operations inside `onBeforeLoad` are asynchronous and treat it as async function, returning the promise.

Reply:
We discussed it F2F with @Comandeer and decided to leave it as is as introducing Promises here will change entire editor initialization flow to async one.